src/Plugin/tmgmt/Translator/ThebigwordTranslator.php

```diff
@@ -0,0 +1,872 @@
+    $required_by->setTimezone(new \DateTimeZone('UTC'));
+    $datetime = (new DrupalDateTime())->format('Y-m-d\TH:i:s');
+    $form_params = [
+      'ProjectId' => $project_id,
+      'RequiredByDateUtc' => $datetime,
+      'SourceLanguage' => $job->getRemoteSourceLanguage(),
+      'TargetLanguage' => $job->getRemoteTargetLanguage(),
+      'FilePathAndName' => 'error-' . (new DrupalDateTime())->format('Y-m-d\TH:i:s') . '.txt',
+      'FileIdToUpdate' => $file_id,
+      'FileState' => $state,
+      'FileData' => base64_encode($message),
+    ];
+    $this->request('file', 'PUT', $form_params);
+    if ($confirm) {
+      $this->confirmUpload($project_id, $state);
+    }
+  }
+
+  /**
+   * Add the prliminary status to the data items.
+   *
+   * @param array $data_items
+   *   Data items.
+   *
+   * @return array
+   *   The data items filled with the preliminart status
+   */
+  private function addPreliminaryStatusToData($data_items) {
+    if (isset($data_items['#text'])) {
+      $data_items['#status'] = TMGMT_DATA_ITEM_STATE_PRELIMINARY;
+      return $data_items;
+    }
+    foreach ($data_items as $key => $data_item) {
+      $data_items[$key] = $this->addPreliminaryStatusToData($data_item);
+    }
+    return $data_items;
+  }
+
+  /**
+   * Retrieve the data of a file in a state.
+   *
+   * @param \Drupal\tmgmt\JobInterface $job
+   *   The Job to which will be added the data.
+   * @param string $state
+   *   The state of the file.
+   * @param string $file_id
+   *   The file ID.
+   *
+   * @throws \Drupal\tmgmt\TMGMTException
+   */
+  private function addFileDataToJob(JobInterface $job, $state, $file_id) {
+    $data = $this->request('file/' . $state . '/' . $file_id);
+    $decoded_data = base64_decode($data['FileData']);
+    $file_data = $this->parseTranslationData($decoded_data);
+    $file_data = $this->addPreliminaryStatusToData($file_data);
+    if ($state == 'TranslatableComplete') {
+      $status = TMGMT_DATA_ITEM_STATE_TRANSLATED;
+    }
+    else {
+      $status = TMGMT_DATA_ITEM_STATE_PRELIMINARY;
+    }
+    $job->addTranslatedData($file_data, [], $status);
+    // Confirm that we download the file.
+    $form_params = [
+      'FileId' => $file_id,
+      'FileState' => $state,
+    ];
+    $this->request('fileinfo/downloaded', 'POST', $form_params);
+
+    if ($status == TMGMT_DATA_ITEM_STATE_PRELIMINARY && $job->getSetting('review')) {
+      foreach ($file_data as $job_item_id) {
+        /** @var JobItem $job_item */
+        $job_item = JobItem::load($job_item_id);
+        $this->sendPreviewUrl($job_item, $file_id, TRUE);
+      }
+    }
+  }
+
+  /**
+   * Confirm all the files uploaded in a project for a state.
+   *
+   * @param int $project_id
+   *   The project ID.
+   * @param string $state
+   *   The state.
+   *
+   * @return array
+   *   The number of confirmed files.
+   *
+   * @throws \Drupal\tmgmt\TMGMTException
+   */
+  private function confirmUpload($project_id, $state) {
```

Comment:
we usually don't use private unless there's a really, really good reason for it.